### PR TITLE
Rustc infer doctest

### PR DESCRIPTION
**Pull request by CarsonV**
More efforts towards #99144. Made some assumptions as I was going along but wanted to get eyes on them now so I can learn before moving on towards completing the rest of the missing doctests.

### CODE REVIEW

**Comment by CarsonV** [comment]
r? jyn514

**Comment by jyn514** [comment]
Sorry for the long wait. I won't have time for reviews for the foreseeable future.

As a note, this has many of the changes from #106574.

r? @Nilstrieb

File: compiler/rustc_infer/src/infer/outlives/verify.rs [hunk 1]
```diff
@@ -288,7 +288,7 @@ impl<'cx, 'tcx> VerifyBoundCx<'cx, 'tcx> {
     ///
     /// It will not, however, work for higher-ranked bounds like:
     ///
-    /// ```compile_fail,E0311
+    /// ```
     /// trait Foo<'a, 'b>
     /// where for<'x> <Self as Foo<'x, 'b>>::Bar: 'x
```
Comment (Nilstrieb) [inline compiler/rustc_infer/src/infer/outlives/verify.rs:293]: The docs say that this shouldn't compile, so there's something wrong here.

**Comment by Dylan-DPC** [comment]
@CarsonV any updates on this?

**Comment by Nilstrieb** [comment]
This was subsumed by #106486, closing it. Thank you for the PR anyways!

**Pull request closed without merging**